setup.py

```diff
@@ -20,6 +20,24 @@
 from setuptools import setup, Extension, find_packages
 from setuptools.command.build_ext import build_ext
 
+default_backend = "lightning_qubit"
+supported_backends = ("lightning_kokkos", "lightning_qubit")
+
+def get_backend():
+    if "CMAKE_ARGS" in os.environ:
+        cmake_args = os.environ["CMAKE_ARGS"].split(" ")
+        arg = [x for x in cmake_args if "PL_BACKEND" in x]
+        backend = arg[0].split("=")[1] if arg else default_backend
+    elif "BACKEND" in os.environ:
+        backend = os.environ.get("BACKEND", default_backend)
+    else:
+        backend = default_backend
+    if backend not in supported_backends:
+        raise ValueError(f"Invalid backend {backend}.")
+    return backend
```

Comment:
Hi @vincentmr, I think we could improve on that. As it is, if someone provides any CMAKE_ARGS, our setup will disregards BACKEND. In this case, `BACKEND="lightning_kokkos" CMAKE_ARGS="-DENABLE_PYTHON=ON" pip install -e .` will install the `lightning_qubit`. And as I can see someone can provide one backend in CMAKE_ARGS and another one in BACKEND (who should win?). I would rather have the backend provided only by the BACKEND variable. What do you think?

Reply:
We could give `PL_BACKEND` precedence, but since this is also a valid CMake option it's legit to pass it in `CMAKE_ARGS`, and hence I think we should look in `CMAKE_ARGS` to get the value if it's there.

Reply:
I agree.